### PR TITLE
Display command options on `--debug`

### PR DESCRIPTION
**Pull request by anvacaru**
- add debug log messages with the options provided to `build` and `prove`.
- add the `to_string()` method for `ProveOptions.`
- update the log handler to use the provided log level; previously, it was hardcoded to `INFO`, preventing the `DEBUG` logs from being displayed.

### CODE REVIEW

File: src/kontrol/options.py [hunk 1]
```diff
@@ -458,6 +458,25 @@ def apply_optimizations(self) -> None:
...
 
+    def to_string(self) -> str:
+        """
+        Generate a string representation of the instance, including attributes from inherited classes.
+
+        The first line collects all attributes that are directly set on the instance.
+        The loop is required to iterate over all parent classes and fetch attributes set by the `default` method.
+
+        :return: String representation of the instance.
+        :rtype: str
```
[the remark this answers is not in the document]
Comment (anvacaru) [reply src/kontrol/options.py:469]: Done!

File: src/kontrol/options.py [hunk 2]
```diff
@@ -458,6 +458,25 @@ def apply_optimizations(self) -> None:
             self.max_depth = 100000
             self.max_iterations = 10000
 
+    def to_string(self) -> str:
```
[the remark this answers is not in the document]
Comment (anvacaru) [reply src/kontrol/options.py:461]: Oh, nice!! Didn't think about this.

[the remark this answers is not in the document]
Comment (tothtamas28) [reply src/kontrol/options.py]: I think so.

File: src/kontrol/kompile.py [hunk 3]
```diff
@@ -132,7 +132,7 @@ def update_kompilation_digest() -> None:
         digest_dict = _read_digest_file(foundry.digest_file)
         digest_dict['kompilation'] = kompilation_digest()
         digest_dict['kontrol'] = VERSION
-        digest_dict['build-options'] = options.to_string()
+        digest_dict['build-options'] = options
```
Comment (tothtamas28) [inline src/kontrol/kompile.py:135]: Are you sure there's an implicit `str` when dumping the JSON?

Comment (anvacaru) [reply src/kontrol/kompile.py:135]: I just tested, I need to wrap it in `str()`.

File: src/kontrol/options.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (anvacaru) [reply src/kontrol/options.py]: I'll try that in an independent PR.

**Review by tothtamas28 — APPROVED** [review]

**Pull request merged**